lib/ohai/plugins/linux/network.rb

```diff
@@ -134,6 +138,25 @@ def parse_routes(family, iface)
     end.compact.flatten
   end
 
+  def ethernet_layer_one(iface)
+    return iface unless ethtool_binary = find_ethtool_binary
+    keys = %w[ Speed Duplex Port Transceiver Auto-negotiation MDI-X ]
+    iface.each_key do |tmp_int|
+      next unless iface[tmp_int][:encapsulation] == 'Ethernet'
+      so = shell_out("#{ethtool_binary} #{tmp_int}")
+      so.stdout.lines do |line|
+        line.chomp!
+        Ohai::Log.debug("Parsing ethtool output: #{line}")
+        line.lstrip!
+        k, v = line.split(': ')
+        next unless keys.include? k
+        k.downcase!
+        v.gsub!(/Mb\/s/, '')
```

Comment:
I agree with everything except the layer two comment. Everything here is a
layer one concept which has been hidden from the Linux kernel in the
interest of simplicity.

Now what about these rspec tests? :D

On Wed, Dec 9, 2015 at 5:22 PM Tim Smith <notifications@github.com> wrote:

> In lib/ohai/plugins/linux/network.rb
> <https://github.com/chef/ohai/pull/680#discussion_r47161637>:
>
> > @@ -134,6 +138,25 @@ def parse_routes(family, iface)
> >      end.compact.flatten
> >    end
> >
> > +  def ethernet_layer_one(iface)
> > +    return iface unless ethtool_binary = find_ethtool_binary
> > +    keys = %w[ Speed Duplex Port Transceiver Auto-negotiation MDI-X ]
> > +    iface.each_key do |tmp_int|
> > +      next unless iface[tmp_int][:encapsulation] == 'Ethernet'
> > +      so = shell_out("#{ethtool_binary} #{tmp_int}")
> > +      so.stdout.lines do |line|
> > +        line.chomp!
> > +        Ohai::Log.debug("Parsing ethtool output: #{line}")
> > +        line.lstrip!
> > +        k, v = line.split(': ')
> > +        next unless keys.include? k
> > +        k.downcase!
> > +        v.gsub!(/Mb\/s/, '')
>
> Also speed should probably get turned into a int so that people can do
> quick comparisons on it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/ohai/pull/680/files#r47161637>.
>